File: 014주차/타겟 넘버/woosteelz.py
answer = 0


def solution(numbers, target):
    dfs(numbers, 0, 0, target)
    return answer


def dfs(arr, _sum, cnt, target):
    global answer
    if cnt == len(arr) and _sum == target:
        answer += 1
        return
    if cnt == len(arr):
        return

    dfs(arr, _sum + arr[cnt], cnt+1, target)
    dfs(arr, _sum - arr[cnt], cnt+1, target)

File: 019주차/기능개발/woosteelz.js
function solution(progresses, speeds) {
    var answer = [];

    while (progresses.length > 0) {
        for (var i = 0; i < progresses.length; i++) {
            progresses[i] = progresses[i] + speeds[i];
        }
        var cnt = 0;
        while (progresses.length > 0 && progresses[0] >= 100) {
            progresses.shift()
            speeds.shift()
            cnt++;
        }
        if (cnt > 0) {
            answer.push(cnt);
        }
    }

    return answer;
}